### PR TITLE
feat: upgrade vested cvx to 1.5

### PR DESCRIPTION
**Pull request by sajanrajdev**
This PR closes issue #142

Upgrades the vestedCVX strat to version 1.5 to include the Chainlink functions. Includes a simulation to ensure that the changes to the strat will work post ugrade.

Run queue with:
```
brownie run scripts/issue/142/upgrade_vestedcvx_strat.py
```
Run execution with:
```
brownie run scripts/issue/142/upgrade_vestedcvx_strat.py main false
```

Run simulation with:
```
brownie run scripts/issue/142/upgrade_vestedcvx_strat.py main false true
```

### CODE REVIEW

**Comment by sajanrajdev** [comment]
Call and events trace:
```
Running 'scripts/issue/142/upgrade_vestedcvx_strat.py::main'...
Transaction sent: 0x48743869f6b54a1efa68289fd817d80b089b92d7bfa00860af54da2da3cdd5ce
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 1
  GovernanceTimelock.queueTransaction confirmed   Block: 14225244   Gas used: 52880 (0.44%)

Dump Directory: data/badger/timelock/upgrade_veCVX_strategy_V1_5/
Transaction sent: 0x35237921502ef89e9367806e78af063c8b4bd9fac54f627a2bcd198586f1f0ee
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 62
  Gnosis Safe.approveHash confirmed   Block: 14225244   Gas used: 46206 (0.39%)

Transaction sent: 0xefae060cff6ec62ddb914fce1d288439815f810ce21edcfcf764b35e8bb2945c
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 180
  Gnosis Safe.approveHash confirmed   Block: 14225245   Gas used: 46206 (0.39%)

Transaction sent: 0x6192b7b2bb3f16f1af67b1b5fef2f3475a6e82d3af485c24b92316444b8f3937
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 0
  Gnosis Safe.approveHash confirmed   Block: 14225246   Gas used: 46206 (0.39%)

Transaction sent: 0xf99a183c31af556371b3627f16f9ce355ca35fb8630020d925d977874a0fe7c3
  Gas price: 0.0 gwei   Gas limit: 99312   Nonce: 63
  Gnosis Safe.execTransaction confirmed   Block: 14225247   Gas used: 61772 (62.20%)

Transaction was Mined
---------------------
Tx Hash: 0xf99a183c31af556371b3627f16f9ce355ca35fb8630020d925d977874a0fe7c3
From: 0x15b8Fe651C268cfb5b519cC7E98bd45C162313C2
To: 0xB65cef03b9B89f99517643226d76e286ee999e77
Value: 0
Function: Gnosis Safe.execTransaction
Block: 14225247
Gas Used: 61772 / 99312 (62.2%)

Events In This Transaction
--------------------------
├── GovernanceTimelock (0x21CF9b77F88Adf8F8C98d7E33Fe601DC57bC0893)
│   └── QueueTransaction
│       ├── txHash: 0xf896cb2fdf34c956f4103f6376c933e20ae59fab1a6613d861e24d5db6d0c397
│       ├── target: 0x20Dce41Acca85E8222D6861Aa6D23B6C941777bF
│       ├── value: 0
│       ├── signature: upgrade(address,address)
│       ├── data: 0x0000000000000000000000003ff634ce65cdb8cc0d569d6d1697c41aa666cea9000000000000000000000000ea563b601c2ce9c67a1f41222856e06aab76d1fa
│       └── eta: 1645467855
│
└── Gnosis Safe (0xB65cef03b9B89f99517643226d76e286ee999e77)
    └── ExecutionSuccess
        ├── txHash: 0x9058d8abbcf6cd3c32ec6a851f4535ece8318cd83e4451555fe49c145c5090bf
        └── payment: 0
```

File: helpers/constants.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+AddressZero = "0x0000000000000000000000000000000000000000"
```
Comment (jorijnsmit) [inline helpers/constants.py:1]: just saying there is `brownie.ZERO_ADDRESS`

Comment (jorijnsmit) [reply helpers/constants.py:1]: also redundant in this pr seems like..?

File: helpers/addresses.py [hunk 2]
```diff
@@ -267,7 +267,7 @@
         "StrategyCvxHelper": "0x9A12A9141363A5B343B781c4951d42E327B89397", # V1.1
         "StrategyCvxCrvHelper": "0x76328277232c97BAf76D23A69015CB478293A048", # V1.1
         "KeeperAccessControl": "0x4fe70eE8fa906D59A88DE5946F114BdbFC410a80",
-        "native.vestedCVX": "0x56df16192D586b288F53852466560b445E21939b", # V1.4
+        "native.vestedCVX": "0xeA563B601C2Ce9c67a1f41222856E06AAB76D1fa", # V1.5
```
Comment (jorijnsmit) [inline helpers/addresses.py:270]: verified source, deployed by the dao, v1.5 strat

**Review by jorijnsmit — APPROVED** [review]
lgtm

File: helpers/constants.py (hunk 1 above)
Comment (sajanrajdev) [reply helpers/constants.py:1]: Ah, didn't know about `brownie.ZERO_ADDRESS` nice! And you are right, not needed here, just copied the whole constants.py file from other repo while I was at it. Will remove the unnecessary ones.

File: helpers/addresses.py (hunk 2 above)
Comment (petrovska-petro) [reply helpers/addresses.py:270]: confirmed ops_deployer3 deployed and [diff](https://github.com/GalloDaSballo/vested-cvx/compare/cvx.1.4...cvx.1.5-rfc.1) introducing keeper methods

**Review by petrovska-petro — APPROVED** [review]
upgrade script looks good !

**Review by GalloDaSballo — COMMENTED** [review]
Code LG

Output from terminal is odd
Screenshot 2022-02-18 at 00 19 07


The calldata doesn't match an upgrade request, can you tell me why?

**Comment by GalloDaSballo** [comment]
Running with
`brownie run scripts/issue/142/upgrade_vestedcvx_strat.py`

Shows the expected Bytecode
Screenshot 2022-02-18 at 00 20 58

**Comment by GalloDaSballo** [comment]
LGTM, let's gooooo!

**Review by dapp-whisperer — APPROVED** [review]
Approving script

**Pull request merged**